scanpy/external/tl/_trimap.py

```diff
@@ -0,0 +1,148 @@
+from ... import logging as logg
+import numpy as np
+import scipy.sparse as scp
+
+
+def trimap(
+    adata,
+    n_dims=2,
+    n_inliers=10,
+    n_outliers=5,
+    n_random=5,
+    metric='euclidean',
+    weight_adj=500.0,
+    lr=1000.0,
+    n_iters=400,
+    init_pos=None,
+    verbose=False,
+    copy=False,
+):
```

Comment:
Please add type hints here. Check other methods for how to do `metric` using `compat.Literal`.

Don’t forget the return type!